src/python/pants/backend/python/providers/pyenv/rules.py

```diff
@@ -0,0 +1,183 @@
+    )
+    if python_to_use is None:
+        raise ValueError(
+            f"Couldn't determine a compatible Interpreter Constraint from {python_setup.interpreter_versions_universe}"
+        )
+
+    which_python_result = await Get(
+        ProcessResult,
+        Process(
+            [pyenv.exe, "latest", "--known", python_to_use],
+            input_digest=pyenv.digest,
+            description=f"Choose specific version for Python {python_to_use}",
+            env={"PATH": env_vars.get("PATH", "")},
+            # Caching the result is OK, since if the user really needs a different patch,
+            # they should list a more precise IC.
+        ),
+    )
+    specific_python = which_python_result.stdout.decode("ascii").strip()
+
+    shim_digest = await Get(
+        Digest,
+        CreateDigest(
+            [
+                FileContent(
+                    "install_python_shim.sh",
+                    dedent(
+                        f"""\
+                        #!/usr/bin/env bash
+                        set -e
+                        DEST=$(realpath {PYENV_NAMED_CACHE}/{specific_python})
+                        if [ ! -d "$DEST" ]; then
+                            export PYENV_ROOT=$(realpath $(mktemp -d -u -p {PYENV_NAMED_CACHE} {specific_python}.XXXXXX))
+                            # export LDFLAGS="$LDFLAGS -Wl,-rpath=PYENV_ROOT/lib"
+                            {pyenv.exe} install {specific_python}
+                            # Removing write perms helps ensure users aren't accidentally modifying Python
+                            # or the site-packages
+                            chmod -R -w "$PYENV_ROOT"/versions/{specific_python}
+                            chmod +w "$PYENV_ROOT"/versions/{specific_python}
+                            ln -s "$PYENV_ROOT"/versions/{specific_python} {PYENV_NAMED_CACHE}/{specific_python}
+                            rm -rf "$PYENV_ROOT"/shims
+                        fi
+                        echo "$DEST"/bin/python
+                        """
+                    ).encode("ascii"),
+                    is_executable=True,
+                )
+            ]
+        ),
+    )
+    digest = await Get(Digest, MergeDigests([shim_digest, pyenv.digest]))
+
+    # NB: We don't cache this process at any level for two reasons:
+    #   1. Several tools (including pex) refer to Python at an absolute path, so a named cache is
+    #   the only way for this to work reasonably well. Since the named cache could be wiped between
+    #   runs (technically during a run, but we can't do anything about that) the
+    #   fastest-yet-still-correct solution is to always run this process and make it bail
+    #   early-and-quickly if the requisite Python already exists.
+    #   2. Pyenv compiles Python using whatever compiler the system is configured to use. Python
+    #   then stores this information so that it cn use the same compiler when compiling extension
```

Comment:
```suggestion
    #   then stores this information so that it can use the same compiler when compiling extension
```

But I wouldn't trigger a new CI run to fix this. Unless you make more changes in this PR, just tack it on to some unrelated future PR.